### PR TITLE
[REF] point_of_sale: unify numpads

### PR DESCRIPTION
**Pull request by vlst-odoo**
At the moment, the point_of_sale module has 3 different implementations of numpads. This is absurd. In this PR we unify them all into a single numpad component.

One of the reasons why there are multiple numpads is because each requires a slightly different set of buttons. The new `Numpad` component provides a simple API which allows the consumer to specify what the buttons should be.

This is done by proving the `buttons` prop, which has the following type:
```js
    buttons: {
        type: Array,
        element: {
            type: Object,
            shape: {
                value: String,
                text: { type: String, optional: true },
                class: { type: String, optional: true },
                disabled: { type: Boolean, optional: true },
            },
        },
    }
```

In this PR we also provide a simple set of helper functions for the tours and adapt all of the existing tests accordingly.
https://github.com/odoo/enterprise/pull/46427
Task: 3470346

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/pos_store.js
Comment (caburj) [inline addons/point_of_sale/static/src/app/store/pos_store.js:1]: get rid of this diff. doesn't seem to be related.

File: addons/point_of_sale/static/tests/tours/Chrome.tour.js
Comment (caburj) [inline addons/point_of_sale/static/tests/tours/Chrome.tour.js:1]: if not related, get rid of this diff.

File: addons/point_of_sale/static/tests/tours/FixedPriceNegativeQty.tour.js
Comment (caburj) [inline addons/point_of_sale/static/tests/tours/FixedPriceNegativeQty.tour.js:1]: this one as well, remove this diff.

File: addons/point_of_sale/static/tests/tours/PaymentScreen.tour.js
Comment (caburj) [inline addons/point_of_sale/static/tests/tours/PaymentScreen.tour.js:1]: If nothing changed here, remove the diff.

File: addons/point_of_sale/static/tests/tours/ReceiptScreen.tour.js
Comment (caburj) [inline addons/point_of_sale/static/tests/tours/ReceiptScreen.tour.js:1]: remove diff if no important change in this file.

File: addons/point_of_sale/static/tests/tours/TicketScreen.tour.js
Comment (caburj) [inline addons/point_of_sale/static/tests/tours/TicketScreen.tour.js:1]: remove diff if nothing important

File: addons/pos_hr/static/tests/tours/PosHrTour.js
Comment (caburj) [inline addons/pos_hr/static/tests/tours/PosHrTour.js:1]: get rid of useless diff in this file.

**Review by caburj — CHANGES_REQUESTED** [review]
if the file is only changed because of formatting, get rid of that diff. It will be helpful for the review.

File: addons/point_of_sale/static/tests/tours/helpers/ProductScreenTourMethods.js [hunk 1]
```diff
@@ -67,32 +67,14 @@ class Do {
      *
      * @param {String} keys space-separated numpad keys
```
Comment (vlst-odoo) [inline addons/point_of_sale/static/tests/tours/helpers/ProductScreenTourMethods.js:68]: adapt this docstring

**Pull request closed without merging**